dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/LocalFileTreeTablePresenter.java

```diff
@@ -506,37 +507,42 @@ private void startPutJob(final Session session,
             final String priority,
             final JobInterruptionStore jobInterruptionStore,
             final TreeItem<Ds3TreeTableValue> remoteDestination) {
-        final Ds3PutJob putJob = new Ds3PutJob(session.getClient(), files, bucket, targetDir, jobInterruptionStore, priority,
-                settingsStore.getProcessSettings().getMaximumNumberOfParallelThreads(),
-                resourceBundle, settingsStore, loggingService, deepStorageBrowserPresenter, dateTimeUtils, remoteDestination);
-        putJob.setOnSucceeded(event -> {
-            LOG.info("BULK_PUT job {} Succeed.", putJob.getJobId());
-            refreshBlackPearlSideItem(remoteDestination);
-        });
-        putJob.setOnFailed(failEvent -> {
-            LOG.info("BULK_PUT job {} Failed.", putJob.getJobId());
-            refreshBlackPearlSideItem(remoteDestination);
-        });
-        putJob.setOnCancelled(cancelEvent -> {
-            try {
-                if (putJob.getJobId() != null) {
-                    session.getClient().cancelJobSpectraS3(new CancelJobSpectraS3Request(putJob.getJobId()));
-
+        try (final Ds3Client client = session.getClient()) {
+            final Ds3PutJob putJob = new Ds3PutJob(client, files, bucket, targetDir, jobInterruptionStore, priority,
+                    settingsStore.getProcessSettings().getMaximumNumberOfParallelThreads(),
+                    resourceBundle, settingsStore, loggingService, deepStorageBrowserPresenter, dateTimeUtils, remoteDestination);
+            putJob.setOnSucceeded(event -> {
+                LOG.info("BULK_PUT job {} Succeed.", putJob.getJobId());
+                refreshBlackPearlSideItem(remoteDestination);
+            });
+            putJob.setOnFailed(failEvent -> {
+                LOG.info("BULK_PUT job {} Failed.", putJob.getJobId());
```

Comment:
This is the possibility that `putJob.getJobId()` will return `null` in the event that the job fails before the job id is returned from black pearl.